v2g-liberty/rootfs/root/appdaemon/tests/test_settings_manager.py

```diff
@@ -276,6 +276,36 @@ def test_upgrade_schedule_settings_initialised(self, settings_manager):
             settings_manager.get("input_boolean.schedule_settings_initialised") is True
         )
 
+    @patch("os.path.exists", lambda _: True)
+    def test_upgrade_fm_url(self, settings_manager):
+        # Arrange
+        saved_settings = json.dumps(
+            {
+                "input_text.fm_host_url": "https://seita.energy",
+            }
+        )
+        with patch("builtins.open", mock_open(read_data=saved_settings)):
+            # Act
+            settings_manager.retrieve_settings()
+        # Assert
+        assert (
+            settings_manager.get("input_text.fm_host_url") == "https://ems.seita.energy"
+        )
+
+    @patch("os.path.exists", lambda _: True)
+    def test_upgrade_fm_url_non_default(self, settings_manager):
```

Comment:
```suggestion
    def test_upgrade_fm_url_non_default(self, settings_manager):
        """Check that the url to other FlexMeasures servers are not affected by Seita's url update from March 2025."""
```

Reply:
Added similar comment and TODO to review if this code can be removed after all users have upgraded.